### PR TITLE
Parse multi-line bold and strikethrough markdown

### PR DESCRIPTION
**Pull request by neil-marcellini**
### Details
In App we want to be able to send multi-line Markdown comments with **bold** or ~strikethrough~ style. Currently, only strikethrough works to send a multi-line comment, and neither works when editing the comment. I have added tests for both styles and I have updated the regexes (regi? 😂 ) to make all the tests pass.

I made the bold Markdown to html regex very similar to the one for strikethrough. Hopefully I didn't break any untested features in that process. Next, I modified the html to Markdown regexes to match everything inside the tags including whitespace.

@parasharrajat, @aimane-chnaif, @marcaaron I would love to have your eyes on this as well so I don't break anything.

### Related Issues
$ https://github.com/Expensify/App/issues/13727

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
Idk what autoQA is, but here's the tests to run.
```
npm test -- __tests__/ExpensiMark-HTML-test.js
```
```
npm test -- __tests__/ExpensiMark-Markdown-test.js
```
2. What tests did you perform that validates your changed worked?
I have tested in the corresponding App PR https://github.com/Expensify/App/pull/14009.

# QA
1. What does QA need to do to validate your changes?
Run the QA in the App PR https://github.com/Expensify/App/pull/14009.
1. What areas to they need to test for regressions?
General Markdown parsing in App?

### CODE REVIEW

**Comment by neil-marcellini** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by marcaaron** [comment]
Not really too great at regex tbh but took a quick look and it LGTM @neil-marcellini

**Review by Julesssss — APPROVED** [review]
I'm also not confident with my regex review, but the test additions look good. Might be worth asking for a review from someone with more Regex knowledge.

**Comment by neil-marcellini** [comment]
@aimane-chnaif hmm that's very odd it works for me. I'm using the latest version of both branches and I ran `npm install` in App before starting the web server? Would you please try again? Maybe you were on an older commit before I fixed the editing behavior?

https://user-images.githubusercontent.com/26260477/210859597-cda17614-567a-4fa7-8a8f-bbf0a7ad2aaf.mov

**Comment by neil-marcellini** [comment]
> @neil-marcellini I found this bug while testing but I think it can be handled in a separate issue since it also happens with normal text (without bold or strikethrough)

Ok great thanks. Please report it in #expensify-bugs in Slack. Do you think this PR is good to go? Once you approve I'll merge it.

**Pull request merged**
